ts/test/session/sending/PendingMessageCache_test.ts

```diff
@@ -0,0 +1,202 @@
+import { expect } from 'chai';
+
+import * as MessageUtils from '../../../session/utils';
+import { TestUtils } from '../../../test/test-utils';
+import { PendingMessageCache } from '../../../session/sending/PendingMessageCache';
+import { PubKey } from '../../../session/types';
+
+// Equivalent to Data.StorageItem
+interface StorageItem {
+  id: string;
+  value: any;
+}
+
+describe('PendingMessageCache', () => {
+  // Initialize new stubbed cache
+  let pendingMessageCacheStub: PendingMessageCache;
+
+  beforeEach(async () => {
+    // Stub out methods which touch the database
+    const storageID = 'pendingMessages';
+    let data: StorageItem = {
+      id: storageID,
+      value: '',
+    };
+
+    TestUtils.stubData('getItemById')
+      .withArgs('pendingMessages')
+      .callsFake(async () => {
+        return data;
+      });
+
+    TestUtils.stubData('createOrUpdateItem').callsFake((item: StorageItem) => {
+      data = item;
+    });
+
+    pendingMessageCacheStub = new PendingMessageCache();
+    await pendingMessageCacheStub.init();
```

Comment:
instead of this we should have a `isReady` variable.

implementation wise we could so something like:
```
const isReady = new Promise(async resolve => {
    await this.init();
    resolve();
});
```